### PR TITLE
tf compressor API refactor

### PR DESCRIPTION
**Pull request by Cjkkkk**

### CODE REVIEW

File: src/sdk/pynni/nni/compression/tensorflow/compressor.py [hunk 1]
```diff
@@ -184,11 +187,13 @@ def _instrument_layer(self, layer, config):
...
 
 
-def _detect_weight_index(layer):
-    index = default_layers.op_weight_index.get(layer.type)
+def _detect_weight_index(op):
+    index = default_layers.op_weight_index.get(op.type)
     if index is not None:
         return index
-    weight_indices = [i for i, op in enumerate(layer.inputs) if op.name.endswith('Variable/read')]
+    # TODO should use "Weight(_\d+)?/read:0$" ?
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/compression/tensorflow/compressor.py]: ?????

File: docs/en_US/Compressor/Overview.md [hunk 2]
```diff
@@ -170,12 +170,30 @@ class YourQuantizer(nni.compression.tensorflow.Quantizer):
...
+        weight is the target weight tensor of a layer
         config is the selected dict object in config_list for this layer
         kwargs contains op, op_types, and op_name
         design your quantizer and return new weight
         """
         return new_weight
+    
+    def quantize_output(self, output, config, **kwargs):
+        """
+        output is the target output tensor of a layer
```
Comment (xuehui1991) [inline docs/en_US/Compressor/Overview.md]: docstring format?

File: docs/en_US/Compressor/Overview.md [hunk 3]
```diff
@@ -170,12 +170,30 @@ class YourQuantizer(nni.compression.tensorflow.Quantizer):
...
+        config is the selected dict object in config_list for this layer
+        kwargs contains op, op_types, and op_name
+        design your quantizer and return new output
+        """
+        return new_output
+
+    def quantize_input(self, *inputs, config, **kwargs):
+        """
+        input is the target input tensor of a layer
+        config is the selected dict object in config_list for this layer
```
Comment (xuehui1991) [inline docs/en_US/Compressor/Overview.md]: wrong doc string format

File: docs/en_US/Compressor/Quantizer.md [hunk 4]
```diff
@@ -28,27 +28,37 @@ In [Quantization and Training of Neural Networks for Efficient Integer-Arithmeti
...
-from nni.compressors.tensorflow import QAT_Quantizer
-config_list = [{ 'q_bits': 8, 'op_types': ['default'] }]
-quantizer = QAT_Quantizer(tf.get_default_graph(), config_list)
-quantizer.compress()
-```
 PyTorch code
 ```python
 from nni.compressors.torch import QAT_Quantizer
-config_list = [{ 'q_bits': 8, 'op_types': ['default'] }]
+model = Mnist()
```
Comment (xuehui1991) [inline docs/en_US/Compressor/Quantizer.md]: add a blank line here.

File: src/sdk/pynni/nni/compression/tensorflow/compressor.py (hunk 1 above)
Comment (Cjkkkk) [reply src/sdk/pynni/nni/compression/tensorflow/compressor.py]: can you be more specific?

File: docs/en_US/Compressor/Quantizer.md (hunk 4 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Quantizer.md]: it is just for discussions about tensorflow compressor API, not a formal pull request

**Pull request closed without merging**